crates/ruff_python_semantic/src/model.rs

```diff
@@ -932,6 +932,29 @@ impl<'a> SemanticModel<'a> {
         None
     }
 
+    /// Return the [`Nodes`] vector of all statements.
+    pub const fn statements(&self) -> &Nodes<'a, Stmt> {
+        &self.statements
+    }
+
+    /// Return the [`NodeId`] corresponding to the given [`Stmt`].
+    #[inline]
+    pub fn statement_id(&self, statement: &Stmt) -> Option<NodeId> {
+        self.statements.node_id(statement)
+    }
+
+    /// Return the [`Stmt]` corresponding to the given [`NodeId`].
+    #[inline]
+    pub fn statement(&self, statement_id: NodeId) -> &'a Stmt {
+        self.statements[statement_id]
+    }
+
+    /// Given a [`Stmt`], return its parent, if any.
+    #[inline]
+    pub fn parent_statement(&self, statement: &'a Stmt) -> Option<&'a Stmt> {
```

Comment:
Should this be `statement_parent` to be consistent with `scope_parent`?

Reply:
Yes, I did this before I changed the prior PR.

Reply:
(Settled on using `parent_scope` and `parent_statement`.)